--- fontforge-2.0.20170731/configure.ac.orig	2017-07-31 10:30:25.294015000 +0200
+++ fontforge-2.0.20170731/configure.ac	2018-06-21 21:57:04.087315481 +0200
@@ -622,7 +622,7 @@
 test x"${i_do_have_cairo}" = xyes && my_libs="${my_libs} ${CAIRO_LIBS}"
 test x"${i_do_have_cairo}" = xyes && my_libs="${my_libs} ${PANGOCAIRO_LIBS}"
 test x"${i_do_have_gui}" = xyes && my_libs="${my_libs} ${PANGO_LIBS}"
-test x"${i_do_have_x}" = xyes && my_libs="${my_libs} ${X_PRE_LIBS} ${X_LIBS} ${X_EXTRA_LIBS}"
+test x"${i_do_have_x}" = xyes && my_libs="${my_libs} ${X_LIBS} ${X_EXTRA_LIBS}"
 test x"${i_do_have_libreadline}" = xyes && my_libs="${my_libs} ${LIBREADLINE_LIBS}"
 test x"${i_do_have_libspiro}" = xyes && my_libs="${my_libs} ${LIBSPIRO_LIBS}"
 my_libs="${my_libs} ${LIBSPIRO_LIBS}"
--- fontforge-2.0.20170731/Makefile.am.orig	2017-07-22 07:00:38.802074006 +0200
+++ fontforge-2.0.20170731/Makefile.am	2018-06-21 22:20:54.687299149 +0200
@@ -66,6 +66,7 @@
 SUBDIRS += collab
 endif
 
+SUBDIRS += plugins
 SUBDIRS += fontforge
 
 if GRAPHICAL_USER_INTERFACE
@@ -90,7 +91,6 @@
 	contrib \
 	doc \
 	nodejs \
-	plugins \
 	po \
 	pycontrib \
 	share \
