--- fontforge-20051205/fontforge/Makefile.dynamic.in.orig	2005-11-11 03:41:01.000000000 +0100
+++ fontforge-20051205/fontforge/Makefile.dynamic.in	2005-12-25 13:34:04.000000000 +0100
@@ -54,7 +54,7 @@
     $(X_PRE_LIBS) $(X_11LIB) $(X_EXTRA_LIBS) @LIBS@ $(STATIC_LIBS) -L/usr/lib -lm
 DLIBS = -rpath $(libdir) ../libgunicode.la @LIBS@ $(STATIC_LIBS) -lm
 
-all: fontforge
+all: fontforge sfddiff
 
 fontforge: main.o $(fontforge_LIBOBJECTS)
 	$(LIBTOOL) --mode=link $(CC) -export-dynamic -o fontforge main.o $(fontforge_LIBOBJECTS) $(LIBS)
