--- fontforge-20050310/configure.in.orig	2005-04-30 16:46:12.479221200 +0200
+++ fontforge-20050310/configure.in	2005-04-30 17:02:34.048000064 +0200
@@ -76,7 +76,7 @@
 AC_SEARCH_LIBS(dlopen, dl) dnl some systems have this in libc (Darwin uses dyld)
 AC_SEARCH_LIBS(inet_aton, resolv) dnl some systems have this in libc
 if test "$gww_uselessiconv" = "no" ; then
- AC_SEARCH_LIBS(libiconv, iconv) dnl linux has iconv in libc, mac,solaris have libiconv in libiconv, and a define to map iconv() to libiconv(), so we can't do the obvious check
+ AC_SEARCH_LIBS(iconv, iconv) dnl linux has iconv in libc, mac,solaris have libiconv in libiconv, and a define to map iconv() to libiconv(), so we can't do the obvious check
 fi
 
 dnl Is there a better way to add a directory to the include path?
